platforms/core-runtime/build-operations-trace/src/main/java/org/gradle/internal/operations/trace/BuildOperationTrace.java

```diff
@@ -216,117 +222,95 @@ public void stop() {
     }
 
     private void write(SerializedOperation operation) {
-        Thread currentThread = Thread.currentThread();
-        ClassLoader previousClassLoader = currentThread.getContextClassLoader();
-        currentThread.setContextClassLoader(JsonOutput.class.getClassLoader());
         try {
-            String json = jsonGenerator.toJson(operation.toMap());
-            try {
-                synchronized (logOutputStream) {
-                    logOutputStream.write(json.getBytes(StandardCharsets.UTF_8));
-                    logOutputStream.write(NEWLINE);
-                    logOutputStream.flush();
-                }
-            } catch (IOException e) {
-                throw UncheckedException.throwAsUncheckedException(e);
+            String json = objectMapper.writeValueAsString(operation.toMap());
```

Comment:
🚀 That would still potentially generate huge strings in memory. Is there an easy way to write directly to `logOutputStream` instead? I was thinking on using the Jackson streaming API but perhaps there's something more direct 🤔

Reply:
This can be replaced by directly writing to the stream, though it’s much less of a problem than the final build operation tree json.

I’ll change this

Reply:
A lot of the manual creation of maps in this code can be replaced now that we use Jackson but I want to keep the scope of this change targeted

Reply:
I can confirm these changes make it possible to capture a build operation trace for `assembleDebug` on `perf-android-large-2` which was impossible before 👍

Reply:
I will fix this in a follow-up PR where I offload writing the build operation trace log file to a separate thread.